internal/monitor/monitor.go

```diff
@@ -144,6 +156,36 @@ func (pm *PowerMonitor) initZones() error {
 	return nil
 }
 
+// collectionLoop handles periodic data collection
+func (pm *PowerMonitor) collectionLoop() {
+	if err := pm.collectData(); err != nil {
+		pm.logger.Error("Failed to collect initial power data", "error", err)
+	}
+
+	if pm.interval > 0 {
+		pm.scheduleNextCollection()
+	}
+}
+
+// scheduleNextCollection schedules the next data collection
+func (pm *PowerMonitor) scheduleNextCollection() {
+	timer := pm.clock.After(pm.interval)
+
+	go func() {
+		select {
+		case <-timer:
+			if err := pm.collectData(); err != nil {
+				pm.logger.Error("Failed to collect power data", "error", err)
+			}
+			pm.scheduleNextCollection()
```

Comment:
call `collectionLoop()` here?

Reply:
I find this "easier" to reason ... 
* `collectionLoop` starts the collection and then `schedules Next` if there is an interval otherwise it exits 
 
*  scheduleNext() schedules the next collection (recursive) and aborts if canceled